### ISSUE TITLE
Print sequence by colour

### ISSUE THREAD

**Issue by rkarlsba** [bambulab/BambuStudio#1005]
We have print by layer or by object, which is fine, but if you have, say, two blue objects and three red ones, it'd be nice to be able to print a group of objects at once, grouped by colour. This way, a way larger amount of objects can be printed in the same run. Would this be hard to add?

**Comment by TheHogarthWorkshop** [comment]
doesn't it already do this?

**Comment by rkarlsba** [comment]
No, it doesn't. If you print three objects where two are colour A and one is colour B and set it to print by object, it will print all by object. What I suggest, is allowing to print by layer where the colour is the same and thus allowing for more objects on the print bed, since there's no need to separate those of the same colour, as in print by layer, as much as those that will be printed separately. I can, of course, print objects of the same colour in different prints to achive the same, but if that's a long print, it'd be better to have it easily automated.

**Comment by rkarlsba** [comment]
<img width="682" alt="image" src="https://user-images.githubusercontent.com/4253147/211102089-d481cb80-5eac-4803-80bf-d86edb1af6d2.png">
Also, the print order is wrong with print by object. Instead of printing by colour, it just chooses a random piece. This makes it flush excessively

**Comment by lanewei120** [comment]
when print by object
you can change the print order by drag the object in the object-list

you can refer to this wiki https://wiki.bambulab.com/en/software/bambu-studio/sequent-print for the details

**Comment by rkarlsba** [comment]
I know, but what I'm suggesting is an extension to this. As the article states, "Print-by-object can be used when we want to print object one by one." I'd like to print object of the same colour in a group and then move to the objects of another colour, so very similar to one-by-one, only that you have a group of reds and then a group of blues and so on and print each of them separate.

**Comment by rkarlsba** [comment]
Also, please note that this is needed since if I use "by object" alone, the needed free area between each object reduces the total object count drastically. This would be a lot better if same-colour-objects could be grouped and printed as one.

**Comment by lanewei120** [comment]
do you mean when using "by object"
the objects in the group with same color will still be printed by layer?

**Comment by rkarlsba** [comment]
What I suggest is that if you have several objects of the same and different colours, grouping those objects into colour groups and printing each group by layer, will both speed up prints lightly and most importantly, make better use available space on the bed. 

Vennlig hilsen 

roy 
-- 
Roy Sigurd Karlsbakk 
(+47) 98013356 
-- 
I all pedagogikk er det essensielt at pensum presenteres intelligibelt. Det er et elementært imperativ for alle pedagoger å unngå eksessiv anvendelse av idiomer med xenotyp etymologi. I de fleste tilfeller eksisterer adekvate og relevante synonymer på norsk.

**Comment by lanewei120** [comment]
thanks for the suggestion

it is a good idea, which will be discussed internally soon

**Comment by speleobrad** [comment]
Any update?

**Comment by lanewei120** [comment]
it is a good feature which it is in our futture plan

however not in recent versions

**Comment by speleobrad** [comment]
Great suggestion - would love to see it automated, but this works too :)
